robot/robot.go

```diff
@@ -113,6 +113,10 @@ type LocalRobot interface {
 
 	// ModuleManager returns the module manager the robot is using.
 	ModuleManager() modif.ModuleManager
+
+	// RemoveOrphanedResources removes the passed in resources from the resource
+	// tree.
```

Comment:
Can we use the internal robot interface resource manager has as well so we don't expose this to users?